### PR TITLE
Check logging output of cockpit-bridge during testing

### PR DESCRIPTION
**Pull request by stefwalter**
* Fix logging to journal of anything written to stderr
 * Send SIGHUP to direct child processes when parent dies

### CODE REVIEW

File: src/common/cockpitpipe.c [hunk 1]
```diff
@@ -932,72 +936,17 @@ calculate_spawn_flags (const gchar **env,
...
-  /* Print all stderr lines as messages */
-  while ((line = memchr (at, '\n', length)) != NULL)
-    {
-      g_printerr ("%.*s\n", (int)(line - at), at);
-      length -= (line - at) + 1;
-      at = line + 1;
-    }
+  /* Send this signal to all direct child processes, when bridge dies */
+#ifdef __linux
+  prctl (PR_SET_PDEATHSIG, 15);
```
Comment (mvollmer) [inline src/common/cockpitpipe.c]: SIGTERM?

**Comment by mvollmer** [comment]
hubbot has found this on r7:

```
Jul 03 08:01:48 m10 cockpit-wrapper[2098]: **
Jul 03 08:01:48 m10 cockpit-wrapper[2098]: GLib-GIO:ERROR:gdbusobjectmanagerserver.c:963:g_dbus_object_manager_server_emit_interfaces_removed: assertion failed (error == NULL): The connection is closed (g-io-error-quark, 18)
```

File: src/common/cockpitpipe.c (hunk 1 above)
Comment (stefwalter) [inline src/common/cockpitpipe.c]: Changed to SIGHUP as noted in the commit message.

**Comment by mvollmer** [comment]
Was it so that messages from cockpit-bridge would appear as coming from cockpit-ws in the journal?

If they now appear as coming from cockpit-bridge, we need to change testlib.py to add "cockpit-bridge" to the syslog ids that we check.

**Comment by mvollmer** [comment]
In any case, we should also check cockpit-bridge messages.  We might have missed them earlier...

**Comment by mvollmer** [comment]
Sorry, rambling.  What about removing cockpit-wrapper and adding cockpit-bridge?

```
diff --git a/test/testlib.py b/test/testlib.py
index 702979e..d4e2ef5 100644
--- a/test/testlib.py
+++ b/test/testlib.py
@@ -515,7 +515,7 @@ systemctl start docker
     def check_journal_messages(self, machine=None):
         """Check for unexpected journal entries."""
         machine = machine or self.machine
-        syslog_ids = [ "cockpit-wrapper", "cockpit-ws" ]
+        syslog_ids = [ "cockpit-ws", "cockpit-bridge" ]
         messages = machine.journal_messages(syslog_ids, 5)
         messages += machine.audit_messages("14") # 14xx is selinux
         all_found = True
```

**Comment by mvollmer** [comment]
> What about removing cockpit-wrapper and adding cockpit-bridge?

Let's see what hubbot thinks about #2465

**Comment by mvollmer** [comment]
> hubbot has found this on r7:

Ahh, hubbot only complains about the "**", the actual message is allowed.

**Comment by stefwalter** [comment]
Fixed all the issues that tracking cockpit-bridge messages in the tests caused.

**Comment by stefwalter** [comment]
Depends on #2489

**Comment by mvollmer** [comment]
> also depends on storaged merge to get tests to pass. Not worth fixing strange stuff deep inside of gdbus running in cockpit-lvm

Could we just ingore everything output by cockpit-wrapper?

**Comment by mvollmer** [comment]
> Could we just ingore everything output by cockpit-wrapper?

Right, that happens already.

**Comment by mvollmer** [comment]
On RHEL 7: `sudo: sorry, you must have a tty to run sudo`

**Comment by stefwalter** [comment]
Ignore RHEL 7 messages.

File: test/check-keys [hunk 2]
```diff
@@ -144,5 +144,16 @@ class TestKeys(MachineCase):
         b.wait_js_func("ph_count_check", "#account-authorized-keys-list div.list-group-item",  1);
 
         self.allow_restart_journal_messages()
+        self.allow_journal_messages('cannot reauthorize identity(s): unix-user:admin ')
+        self.allow_journal_messages('Error executing command as another user: Not authorized'
```
Comment (mvollmer) [inline test/check-keys]: Syntax error: missing ')' at end.

File: test/check-keys [hunk 3]
```diff
@@ -144,5 +144,16 @@ class TestKeys(MachineCase):
...
+        self.allow_journal_messages('This incident has been reported.')
+        self.allow_journal_messages('authorized_keys is not a public key file.')
+        self.allow_journal_messages('Missing callback called fullpath = /home/user/.ssh/authorized_keys')
+        self.allow_journal_messages('')
+        self.allow_journal_messages('cannot reauthorize identity(s): unix-user:admin ')
```
Comment (mvollmer) [inline test/check-keys]: need to escape the parens:
```
'cannot reauthorize identity\\(s\\): unix-user:admin '
```

File: test/check-journal [hunk 4]
```diff
@@ -208,5 +208,6 @@ ExecStart=/bin/sh -c 'for s in $(seq 10); do echo SLOW; sleep 0.1; done; sleep 1
                      ])
 
         self.allow_restart_journal_messages()
+        self.allow_journal_messages(".*Failed to get realtime timestamp: Cannot assign requested address.*")
```
Comment (mvollmer) [inline test/check-journal]: This needs to go to the top of the test, since we will end the test early on line 177 on RHEL 7.

**Pull request closed without merging**